### ISSUE TITLE
Url configuration redirect url wilcard not saved properly

### ISSUE THREAD

**Issue by sagarchoudhary96** [supabase/supabase#13636]
## Describe the bug

i'm trying to add a wildcard for redirectUrI in supabase auth dashboard, `http://localhost:3000{,/**}`:

[Example with the tool mentioned in subabase doc](https://www.digitalocean.com/community/tools/glob?comments=true&glob=http%3A%2F%2Flocalhost%3A3000%7B%2C%2F%2A%2A%7D&matches=false&tests=http%3A%2F%2Flocalhost%3A3000&tests=http%3A%2F%2Flocalhost%3A3000%2F&tests=http%3A%2F%2Flocalhost%3A3000%2F%3Ftest%3Dtest&tests=http%3A%2F%2Flocalhost%3A3000%2Ftest-test%3Ftest%3Dtest&tests=http%3A%2F%2Flocalhost%3A3000%2Ftest%2Ftest%3Ftest%3Dtest)

when i add it to the dashboard it is getting saved as two url.
<img width="950" alt="Screenshot 2023-04-12 at 12 49 43 PM" src="https://user-images.githubusercontent.com/16102594/231381777-5bf88e5c-e9e9-4be1-91b2-c835038b3b12.png">

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to 'dashboard > auth  > url-configuration'
2. try adding `http://localhost:3000{,/**}`

## Expected behavior

it should be added as one url

**Comment by MildTomato** [comment]
@kangmingtay @J0 
we have an issue here with keeping urls in a comma separated string.

**Comment by J0** [comment]
Oh the URI_ALLOW_LIST environment variable which maps to the list of valid redirect urls lives on the server and we perform checks there.  

By

> separate the string on client, or if even the gotrue middleware can?

Were y'all hoping to pass the uri list to the server on a parameter in the supabase-js client library method which can then be split by either the middleware on GoTrue or the Supabase-js lib? Taking a guess here, because I'm not fully certain.

Do y'all mind also sharing a high level overview of the issue with storing comma separated string? Would be helpful in allowing us to explore alternatives.

**Comment by MildTomato** [comment]
@J0 they are using the redirect `"http://localhost:3000{**,/**}"` but the comma in it forces the FE to split the comma separated string wrong.

I assume on server you are just doing something like value.split(',') ?

**Comment by kangmingtay** [comment]
@MildTomato we're not doing something like redirects.split(','), gotrue is able to parse `"http://localhost:8000,http://localhost:3000{**/**}"` into `["http://localhost:8000", "http://localhost:3000{**/**}"]` so we just need this to be supported on the frontend

**Comment by kangmingtay** [comment]
@MildTomato yeah you're right, it's one of the disadvantages of the library we're currently using to read the environment file for gotrue
